Improves functions data & signature encode methods

pgev opened the pull request:
- Replaces web3.eth.abi.encodeFunctionCall -s by
    MyContract.methods.MyMethod(...).encodeABI() -s.
  - Replaces web3.eth.abi.encodeFunctionSignature() -s by
    MyContract.jsonInterface.abi.methods.MyMethod.signature -s.
  - generateSwapOwnerData and generateCoGatewayRedeemFunctionData
    functions explicitly use corresponding functions' abis as contracts
    of those functions are out of the current project.
  - Updates all (almost) modules within package.json, adjusts
    package-lock.json, fixes versions (no ^) of modules within
    package.json.
  - Increases Truffle's solc version from '0.5.0' to '0.5.8'.
  - Fixes typo: DelayedRedcoveryModuleCreated: removes `d`.
  - Fixes typo: RECOERY_MODULE_DOMAIN_SEPARATOR_TYPEHASH: add `V`.

Resolves: #184

Commit:
Improves functions data & signature encode methods

  - Replaces web3.eth.abi.encodeFunctionCall -s by
    MyContract.methods.MyMethod(...).encodeABI() -s.
  - Replaces web3.eth.abi.encodeFunctionSignature() -s by
    MyContract.jsonInterface.abi.methods.MyMethod.signature -s.
  - generateSwapOwnerData and generateCoGatewayRedeemFunctionData
    functions explicitly use corresponding functions' abis as contracts
    of those functions are out of the current project.
  - Updates all (almost) modules within package.json, adjusts
    package-lock.json, fixes versions (no ^) of modules within
    package.json.
  - Increases Truffle's solc version from '0.5.0' to '0.5.8'.
  - Fixes typo: DelayedRedcoveryModuleCreated: removes `d`.
  - Fixes typo: RECOERY_MODULE_DOMAIN_SEPARATOR_TYPEHASH: add `V`.

Resolves: #184

abhayks1 commented:
Reviewing 🚀

abhayks1 commented on package.json:
```diff
@@ -41,28 +41,24 @@
+        "ganache-cli": "6.4.3",
+        "npm-run-all": "4.1.5",
+        "solium": "1.2.4",
+        "web3": "1.0.0-beta.51",
```
We can update it to 1.0.0-beta.55 if no major test issue.
https://www.npmjs.com/package/web3

pgev replied:
Sure. Could you please create a ticket for that. Thx.

pgev replied:
https://github.com/OpenST/openst-contracts/issues/206

abhayks1 reviewed (dismissed):
Nice work 👍 💯 
Added a small comment. It will be nice if that's handled as we are updating package.json npm versions anyways. Let me know if there are test failures 🔴

Commit:
Updates package.json

  - Moves the following dependencies to devDependencies:
    - openzeppelin-solidity
    - solc
    - truffle
  - Updates truffle version from 5.0.17 -> 5.0.18.

schemar commented:
Checking 🦑

schemar reviewed (approved):
Nice clean-up 🐴
